challenges/08-coding-interview-prep/rosetta-code.json

```diff
@@ -10506,6 +10506,1096 @@
+        },
+        {
+          "text": "'<code>fnOrdinalForm(10)</code> should return a string.'",
+          "testString":
+            "assert(typeof fnOrdinalForm(10) == 'string','<code>fnOrdinalForm(10)</code> should return a string.');"
+        },
+        {
+          "text":
+            "'<code>fnOrdinalForm(10)</code> should return <code>\"10\\'th\"</code>.'",
+          "testString":
+            "assert.equal(fnOrdinalForm(10),\"10'th\",'<code>fnOrdinalForm(10)</code> should return <code>\"10\\'th\"</code>.');"
+        },
+        {
+          "text":
+            "'<code>fnOrdinalForm(874)</code> should return <code>\"874\\'th\"</code>.'",
+          "testString":
+            "assert.equal(fnOrdinalForm(874),\"874'th\",'<code>fnOrdinalForm(874)</code> should return <code>\"874\\'th\"</code>.');"
+        },
+        {
+          "text":
+            "'<code>fnOrdinalForm(579)</code> should return <code>\"579\\'th\"</code>.'",
+          "testString":
+            "assert.equal(fnOrdinalForm(579),\"579'th\",'<code>fnOrdinalForm(579)</code> should return <code>\"579\\'th\"</code>.');"
+        },
+        {
+          "text":
+            "'<code>fnOrdinalForm(1001)</code> should return <code>\"1001\\'st\"</code>.'",
+          "testString":
+            "assert.equal(fnOrdinalForm(1001),\"1001'st\",'<code>fnOrdinalForm(1001)</code> should return <code>\"1001\\'st\"</code>.');"
+        },
+        {
+          "text":
+            "'<code>fnOrdinalForm(1002)</code> should return <code>\"1002\\'nd\"</code>.'",
+          "testString":
+            "assert.equal(fnOrdinalForm(1002),\"1002'nd\",'<code>fnOrdinalForm(1002)</code> should return <code>\"1002\\'nd\"</code>.');"
+        },
+        {
+          "text":
+            "'<code>fnOrdinalForm(36003)</code> should return <code>\"36003\\'rd\"</code>.'",
+          "testString":
+            "assert.equal(fnOrdinalForm(36003),\"36003'rd\",'<code>fnOrdinalForm(36003)</code> should return <code>\"36003\\'rd\"</code>.');"
+        }
+      ],
+      "id": "5a23c84252665b21eecc7f3c",
+      "challengeType": 5,
+      "releasedOn": "July 28, 2018",
+      "files": {
+        "indexjs": {
+          "key": "indexjs",
+          "ext": "js",
+          "name": "index",
+          "contents": ["function fnOrdinalForm (n) {", "  // Good luck!", "}"],
+          "head": [],
+          "tail": []
+        }
+      }
+    },
+    {
+      "title": "N-queens problem",
+      "description": [
+        "Write a function to solve the <a href=\"http://rosettacode.org/wiki/WP:Eight_queens_puzzle\">eight queens puzzle</a>. It should take the number of rows and columns as parameters. It should return all the possible solutions for the given board size in the form of a 2D array. Each element of this array should be of length equal to the number of rows. The i<sup>th</sup> element should be the position of the queen in the i<sup>th</sup> row."
```

Comment:
`"Write a function to solve the <a href=\"http://rosettacode.org/wiki/WP:Eight_queens_puzzle\">eight queens puzzle</a>. It should take the number of rows and columns as parameters. It should return all the possible solutions for the given board size in the form of a 2D array. The length of each element of this array should be equal to the total number of rows. The i<sup>th</sup> element should be the position of the queen in the i<sup>th</sup> row."`